dsl/src/commonMain/kotlin/com/freeletics/flowredux/dsl/FlowReduxStateMachine.kt

```diff
@@ -15,10 +15,12 @@ import kotlinx.coroutines.sync.withLock
 @FlowPreview
 @ExperimentalCoroutinesApi
 abstract class FlowReduxStateMachine<S : Any, A : Any>(
-    private var initialStateSupplier: () -> S,
+    initialStateSupplier: () -> S,
     private val logger: FlowReduxLogger? = null
 ) : StateMachine<S, A> {
 
+    val initialState: S by lazy(LazyThreadSafetyMode.NONE, initialStateSupplier)
```

Comment:
So initial state will be kept forever. That ensures that it is always the same initial state for all `state:Flow<MyState>` collectors. Also with making this public, `FlowReduxStateMachine.asState()` Compose extension can access it and return type is `State<MyState>` instead of nullable  `State<MyState?>` as it serves as the initial state in `produceState {...}`.